apitools/gen/util.py

```diff
@@ -365,6 +367,30 @@ def _NormalizeDiscoveryUrls(discovery_url):
     ]
 
 
+def _Gunzip(gzipped_content):
+    """Returns gunzipped content from gzipped contents."""
+    f = tempfile.NamedTemporaryFile(suffix='gz', mode='w+b', delete=False)
+    try:
+        f.write(gzipped_content)
+        f.close()  # force file synchronization
+        with gzip.open(f.name, 'rb') as h:
+            decompressed_content = h.read()
+        return decompressed_content
+    finally:
+        os.unlink(f.name)
+
+
+def _GetURLContent(url):
+    """Download and return the content of URL."""
+    response = urllib_request.urlopen(url)
+    encoding = response.info().get('Content-Encoding')
+    if encoding == 'gzip':
+        content = _Gunzip(response.read())
+    else:
```

Comment:
would be nice to also add deflate support :)

Reply:
IIUC, it also works with deflate: I just removed the branch as it just repeated what the default branch did. Or am I missing something?

Reply:
I expected some code like, https://github.com/httplib2/httplib2/blob/master/python2/httplib2/__init__.py#L521